transactor/src/main/scala/io/mediachain/transactor/RpcService.scala

```diff
@@ -30,13 +30,11 @@ object RpcService {
     logger.info(s"Connecting to cluster at $cluster...")
     client.connect(cluster)
 
-    implicit val ec = ExecutionContext.global
-    // todo: dynamic
-    val executor = Executors.newFixedThreadPool(4)
-    val datastoreConfig = DynamoDatastore.Config(
-      "Mediachain",
-      Some("http://localhost:8000")
-    )
+    val threads = Math.max(4, Runtime.getRuntime.availableProcessors)
+    implicit val ec = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(threads))
```

Comment:
The execution context is for the futures and rpc service main body.
The extra executor is used by the streaming parts of the service.
I think we might be able to use the same executor for both, but I am not clear how that code works so I took the safe approach and kept them separate.